lib/chef/azure/helpers/shared.rb

```diff
@@ -88,4 +89,36 @@ def report_status_to_azure (message, status_type)
       AzureExtensionStatus.log(@azure_status_file, message, status_type)
     end
   end
+
+  module DeleteNode
+    include ChefAzure::Shared
+
+    def delete_node
+      Chef::Log.info "Inside delete node call"
+      begin
+        Chef::Config.from_file("#{bootstrap_directory}/client.rb")
+
+        unless Chef::Config[:node_name]
+          ohai = Ohai::System.new
+          ohai.all_plugins
+          Chef::Config[:node_name] = ohai[:fqdn] || ohai[:machinename] || ohai[:hostname]

```

Comment:
Can't you use Chef::ConfigFetcher instead? This should get the actual config that chef-client is truly using, and you can just use Chef::Config[node_name].
Alternatively, what about using Chef::Rest? Can you use a client_key to make a rest call for the node_name?